src/tls13.c

```diff
@@ -9473,6 +9473,13 @@ int DoTls13HandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
         break;
     }
 
+#if defined(WOLFSSL_ASYNC_CRYPT) || defined(WOLFSSL_ASYNC_IO)
+    /* if async, offset index so this msg will be processed again */
+    if ((ret == WC_PENDING_E || ret == OCSP_WANT_READ) && *inOutIdx > 0) {
+        *inOutIdx -= HANDSHAKE_HEADER_SZ;
+    }
+#endif
```

Comment:
Maybe just add a comment that this has to come before the `wolfSSL_connect_TLSv13` call.

Reply:
I added a comment before `wolfSSL_connect_TLSv13` instead, I think it's more relevant there. What do you think?

Reply:
you're right. added, even if without context it looks pedantic